src/common/service-discovery.js

```diff
@@ -1,68 +1,32 @@
-import Promise from 'bluebird';
-import Etcd from 'node-etcd';
 import { logger } from './logging';
 import config from './config';
 
-// Reuse a singleton instance of the etcd client for all operations
-let etcdClient = null;
-
-/**
- * Get the etcd client.
- */
-function getEtcdClient() {
-  if (etcdClient !== null) {
-    return etcdClient;
-  }
-
-  let etcdConfig = config.get('etcd');
-  logger.log('debug', `Using etcd at ${etcdConfig.ip}:${etcdConfig.port} for service discovery`);
-  
-  let client = new Etcd(`http://${etcdConfig.ip}:${etcdConfig.port}`);
-  Promise.promisifyAll(client);
-  etcdClient = client;
-  return etcdClient;
-}
+let registry = {
```

Comment:
I may be missing something here, but there are cases where we don't use the default ports so we shouldn't hardcode them.  Again, I could be missing something in the new setup.